### PR TITLE
all the things

### PR DESCRIPTION
**Pull request by stefanpenner**
- use caching thats built into rollup
- use es6 in node
- more tests
- should be feature parity?
- stable output
- node 10 support
- windows support

### CODE REVIEW

**Comment by stefanpenner** [comment]
@chadhietala r?

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import fs from 'fs-extra';
+import Plugin from 'broccoli-plugin';
+import md5Hex from 'md5-hex';
+import path from 'path';
+import { default as _debug } from 'debug';
```
Comment (chadhietala) [inline src/index.js]: Use heimdal

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+const debug = _debug('broccoli-rollup');
+
+export default class Rollup extends Plugin {
+  constructor(node, options = {}) {
+    super([node], options);
+
+    this._persistentOutput = true;
+    this.rollupOptions  = options.rollup || {};
+    this.configPath = options.configPath;
```
Comment (chadhietala) [inline src/index.js]: Delete

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      // hit
+    } else {
+      debug('cache miss, write to: %s', fullPath);
+      fs.writeFileSync(fullPath, content);
+      this._fileToChecksumMap[fullPath] = next; // update map
+    }
+  }
+
+  _loadOptions() {
+    // TODO: support rollup config files
```
Comment (chadhietala) [inline src/index.js:41]: Add `this._lastBundle` back into the options.

File: tests/index.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+          .to.equal('var add = x => x + x;\n\nvar index = add(1);\n\nexport default index;');
+      });
+
+      describe('stability', function(){
+        it('is stable on idempotent rebuild', async function() {
+          let { directory } = await pipeline.build();
+
+          let beforeStat = fs.statSync(directory + '/out.js');
+
+          await new Promise(resolve => setTimeout(resolve, 1000));
```
Comment (chadhietala) [inline tests/index.js:107]: Leave a comment for future travelers.

File: src/index.js (hunk 3 above)
Comment (stefanpenner) [reply src/index.js:41]: addressed

File: package.json [hunk 5]
```diff
@@ -2,27 +2,50 @@
...
+    "build:serve": "broccoli serve",
+    "test": "npm run build && mocha dist/tests",
+    "test:fast": "mocha dist/tests",
+    "test:debug": "npm run build && mocha debug dist/tests"
   },
   "keywords": [
     "broccoli"
   ],
-  "author": "Chad Hietala <chadhietala@gmail.com>",
+  "author": "Chad Hietala <chadhietala@gmail.com> & Stefan Penner <stefan.penner@gmai.com>",
```
Comment (chadhietala) [inline package.json]: Fix your email.

File: package.json [hunk 6]
```diff
@@ -2,27 +2,50 @@
...
+    "chai-files": "^1.4.0",
+    "es6-promise": "^3.2.1",
+    "fixturify": "^0.3.0",
+    "mocha": "^2.5.3",
+    "regenerator-runtime": "^0.9.5",
+    "walk-sync": "^0.3.1"
+  },
+  "dependencies": {
+    "broccoli-plugin": "^1.2.1",
+    "debug": "^2.2.0",
```
Comment (chadhietala) [inline package.json]: Is this needed for `mocha debug`, otherwise remove.

**Comment by chadhietala** [comment]
Just 2 things otherwise LGTM.

**Pull request merged**